File: translate_data.js
import fs from 'fs';
import path from 'path';
import translate from 'google-translate-api-x';

const DATA_DIR = path.join(process.cwd(), 'public', 'data');
const BATCH_SIZE = 5;

async function translateText(text) {
  if (!text || typeof text !== 'string') return text;
  // If it's just a file name or something very short that shouldn't be translated, skip
  if (text.endsWith('.pdf') || text.endsWith('.json') || text.endsWith('.jpg')) return text;
  
  try {
    const res = await translate(text, { to: 'es' });
    return res.text;
  } catch (err) {
    console.error('Translation error:', err);
    return text;
  }
}

async function translateArray(arr) {
  if (!Array.isArray(arr)) return arr;
  const translated = [];
  for (const item of arr) {
    if (typeof item === 'string') {
      translated.push(await translateText(item));
    } else if (typeof item === 'object') {
      translated.push(await translateObject(item));
    } else {
      translated.push(item);
    }
  }
  return translated;
}

async function translateObject(obj) {
  if (!obj || typeof obj !== 'object') return obj;
  if (Array.isArray(obj)) return translateArray(obj);

  const newObj = {};
  for (const [key, value] of Object.entries(obj)) {
    // Keys to translate
    if (['title', 'description', 'subtitle', 'features', 'applications', 'name', 'category'].includes(key)) {
      if (typeof value === 'string') {
        newObj[key] = await translateText(value);
      } else if (Array.isArray(value)) {
        newObj[key] = await translateArray(value);
      } else {
        newObj[key] = value;
      }
    } else {
      // Don't translate keys like 'file', 'id', 'primary_image', etc.
      if (typeof value === 'object') {
         newObj[key] = await translateObject(value);
      } else {
         newObj[key] = value;
      }
    }
  }
  return newObj;
}

async function processFile(filePath) {
  try {
    const content = fs.readFileSync(filePath, 'utf-8');
    let data;
    try {
       data = JSON.parse(content);
    } catch(e) { return; } // skip invalid json
    
    console.log(`Translating: ${path.basename(filePath)}`);
    const translatedData = await translateObject(data);
    
    // Quick string replace for Elite South Texas -> Elite Mexico in case it was missed
    let newContent = JSON.stringify(translatedData, null, 2);
    newContent = newContent.replace(/Elite South Texas/g, 'Elite Mexico');
    newContent = newContent.replace(/Elite Sur de Texas/gi, 'Elite Mexico');
    
    fs.writeFileSync(filePath, newContent, 'utf-8');
  } catch (err) {
    console.error(`Failed to process ${filePath}:`, err);
  }
}

async function main() {
  const files = fs.readdirSync(DATA_DIR).filter(f => f.endsWith('.json'));
  
  // process in small batches to avoid rate limits
  for (let i = 0; i < files.length; i += BATCH_SIZE) {
    const batch = files.slice(i, i + BATCH_SIZE);
    await Promise.all(batch.map(f => processFile(path.join(DATA_DIR, f))));
    console.log(`Processed batch ${Math.floor(i/BATCH_SIZE) + 1} of ${Math.ceil(files.length/BATCH_SIZE)}`);
    // sleep to prevent rate limiting
    await new Promise(r => setTimeout(r, 2000));
  }
  
  console.log("Translation complete!");
}

main();

File: translate_data.py
import json
import os
import glob
from deep_translator import GoogleTranslator

DATA_DIR = os.path.join(os.getcwd(), 'public', 'data')

def translate_text(text):
    if not text or not isinstance(text, str):
        return text
    if text.endswith('.pdf') or text.endswith('.json') or text.endswith('.jpg') or text.endswith('.png'):
        return text
    if len(text.strip()) == 0:
        return text
    try:
        translated = GoogleTranslator(source='en', target='es').translate(text)
        return translated if translated else text
    except Exception as e:
        print(f"Error translating text: {e}")
        return text

def translate_list(lst):
    return [translate_object(item) if isinstance(item, (dict, list)) else translate_text(item) if isinstance(item, str) else item for item in lst]

def translate_object(obj):
    if isinstance(obj, list):
        return translate_list(obj)
    if not isinstance(obj, dict):
        return obj

    new_obj = {}
    for key, value in obj.items():
        if key in ['title', 'description', 'subtitle', 'features', 'applications', 'name', 'category', 'content']:
            if isinstance(value, str):
                new_obj[key] = translate_text(value)
            elif isinstance(value, list):
                new_obj[key] = translate_list(value)
            else:
                new_obj[key] = value
        else:
            if isinstance(value, dict) or isinstance(value, list):
                new_obj[key] = translate_object(value)
            else:
                new_obj[key] = value
    return new_obj

def process_file(filepath):
    try:
        with open(filepath, 'r', encoding='utf-8') as f:
            data = json.load(f)
            
        print(f"Translating: {os.path.basename(filepath)}")
        translated_data = translate_object(data)
        
        # Replace Elite South Texas -> Elite Mexico
        content_str = json.dumps(translated_data, indent=2, ensure_ascii=False)
        content_str = content_str.replace("Elite South Texas", "Elite Mexico")
        content_str = content_str.replace("Elite Sur de Texas", "Elite Mexico")
        content_str = content_str.replace("Elite Sur de Texas", "Elite Mexico")
        
        with open(filepath, 'w', encoding='utf-8') as f:
            f.write(content_str)
    except Exception as e:
        print(f"Failed to process {filepath}: {e}")

def main():
    files = glob.glob(os.path.join(DATA_DIR, "*.json"))
    
    # We want to translate all files that are still in English.
    # We can check if "Elite South Texas" is in the file, OR just re-translate
    # But since translating takes time, let's skip files that already have "Elite Mexico" in them
    # Wait, the previous translation script ALSO replaced Elite South Texas with Elite Mexico.
    # So if it has Elite Mexico, it was probably translated!
    
    files_to_translate = []
    for file in files:
        if os.path.basename(file) in ['catalog.json', 'gallery.json', 'index.json', 'contact-us.json', 'resources.json']:
            continue
        try:
            with open(file, 'r', encoding='utf-8') as f:
                content = f.read()
                if "Elite South Texas" in content:
                    files_to_translate.append(file)
        except:
            pass
            
    print(f"Found {len(files_to_translate)} files that still need translation.")
    
    for file in files_to_translate:
        process_file(file)
        
    print("Translation complete!")

if __name__ == "__main__":
    main()
